apps/spruce/src/pages/image/OperatingSystemTable/index.tsx

```diff
@@ -0,0 +1,117 @@
+import { useMemo, useRef, useState } from "react";
+import { useQuery } from "@apollo/client";
+import {
+  useLeafyGreenTable,
+  LGColumnDef,
+  ColumnFiltersState,
+  PaginationState,
+} from "@leafygreen-ui/table";
+import { BaseTable } from "components/Table/BaseTable";
+import { DEFAULT_PAGE_SIZE } from "constants/index";
+import { useToastContext } from "context/toast";
+import {
+  OsInfo,
+  ImageOperatingSystemQuery,
+  ImageOperatingSystemQueryVariables,
+} from "gql/generated/types";
+import { IMAGE_OPERATING_SYSTEM } from "gql/queries";
+
+type OperatingSystemTableProps = {
+  imageId: string;
+};
+
+export const OperatingSystemTable: React.FC<OperatingSystemTableProps> = ({
+  imageId,
+}) => {
+  const dispatchToast = useToastContext();
+
+  const [pagination, setPagination] = useState<PaginationState>({
+    pageIndex: 0,
+    pageSize: DEFAULT_PAGE_SIZE,
+  });
+  const [columnFilters, setColumnFilters] = useState<ColumnFiltersState>([]);
+  const { data: osData, loading } = useQuery<
+    ImageOperatingSystemQuery,
+    ImageOperatingSystemQueryVariables
+  >(IMAGE_OPERATING_SYSTEM, {
+    variables: {
+      imageId,
+      opts: {
+        page: pagination.pageIndex,
+        limit: pagination.pageSize,
+        name: columnFilters.find((filter) => filter.id === "name")
+          ?.value as string,
+      },
+    },
+    onError(err) {
+      dispatchToast.error(
+        `There was an error loading operating system information: ${err.message}`,
+      );
+    },
+  });
+
+  const operatingSystemInfo = useMemo(
+    () => osData?.image?.operatingSystem.data ?? [],
+    [osData?.image?.operatingSystem.data],
+  );
+
+  const numTotalItems = useMemo(
+    () =>
+      osData?.image?.operatingSystem.filteredCount ??
+      osData?.image?.operatingSystem.totalCount,
+    [
+      osData?.image?.operatingSystem.filteredCount,
+      osData?.image?.operatingSystem.totalCount,
+    ],
+  );
```

Comment:
If I'm not mistaken it seems like these `useMemo`s have no impact. Basically they're saying "if the data value changes, update the variable value" which feels... tautological

Reply:
oh yeah I'm not sure why we started doing that... I removed them all around here

Reply:
Ok I kinda feel like I had Friday brain when I made this comment since the useMemo would still do *something*, but I think removing them is fine too for simplicity's sake since I don't anticipate this page having performance issues.